Sources/SwiftParserDiagnostics/LexerDiagnosticMessages.swift

```diff
@@ -325,6 +325,50 @@ extension SwiftSyntax.TokenDiagnostic {
         changes.append(.replaceLeadingTrivia(token: nextToken, newTrivia: []))
       }
       return [FixIt(message: .removeExtraneousWhitespace, changes: changes)]
+    case .spaceAtStartOfRegexLiteral:
+      var ancestor = Syntax(token)
+      while ancestor.kind != .regexLiteralExpr, let parent = ancestor.parent {
+        ancestor = parent
+      }
+
+      let regexLiteral = ancestor.cast(RegexLiteralExprSyntax.self)
+      let regexText = regexLiteral.regex.text
+      let lastIndex = regexText.index(before: regexText.endIndex)
+      let escapedRegexText: String =
+        if regexText.startIndex != lastIndex && regexText[lastIndex].isWhitespace {
+          #"\\#(regexText[..<lastIndex])\ "#
+        } else {
+          "\\" + regexText
+        }
```

Comment:
swift-syntax needs to be able to build with a Swift 5.7 compiler so we can’t use if expressions yet.